src/edu/psu/sweng500/team8/gui/CellGUI.java

```diff
@@ -0,0 +1,513 @@
+public class CellGUI extends JPanel {
+
+	private GameSession gameSession;
+	private JPanel numberInputCell;
+	private JTextField numberInputField;
+
+	private JPanel pencilMarkInputCell;
+
+	private static int NUMBER_SIZE = 16;
+	private static int CELL_SIZE = 54;
+	private static final Border SELECTED_BORDER = BorderFactory
+			.createLineBorder(Color.blue, 3);
+	private static final Border DEFAULT_BORDER = BorderFactory
+			.createLineBorder(Color.black, 1);
+	private Cell cell;
+
+	private MouseAdapter mouseAdapter;
+	
+	/* Pencil Mark Display */
+	private JPanel pencilMarkDisplayCell;
+	private JLabel[][] pencilMarkDisplay;
+
+	public CellGUI() {
+
+		setMaximumSize(new Dimension(CELL_SIZE, CELL_SIZE));
+		setMinimumSize(new Dimension(CELL_SIZE, CELL_SIZE));
+		setPreferredSize(new Dimension(CELL_SIZE, CELL_SIZE));
+
+		setBorder(new LineBorder(new Color(0, 0, 0), 0));
+		setLayout(new java.awt.GridBagLayout());
+
+		initPencilMarkDisplayCell();
+		initNumberInputCell();
+		initPencilMarkInputCell();
+	}
+
+	public void populatePencilMark(Cell cell, GameSession gameSession) {
+
+		this.numberInputCell.setVisible(false);
+		this.pencilMarkDisplayCell.setVisible(false);
+		this.pencilMarkInputCell.setVisible(true);
+
+		pencilMarkInputCell.removeAll();
+		if (cell.hasNumber()) {
+			pencilMarkInputCell.setLayout(new BorderLayout());
+			pencilMarkInputCell.setBorder(null);
+			pencilMarkInputCell.add(buildReadOnlyTextField(cell));
+		} else {
+			pencilMarkInputCell.setLayout(new GridBagLayout());
+			pencilMarkInputCell.setBorder(DEFAULT_BORDER);
+			pencilMarkInputCell.add(buildPencilMarkInputField(cell));
+		}
+	}
+
+	public void populate(Cell cell, GameSession gameSession, boolean isRefresh, FocusAdapter focusAdapter, MouseAdapter mouseAdapter) {
+
+		this.gameSession = gameSession;
+		this.cell = cell;
+		this.mouseAdapter = mouseAdapter;
+
+		/* Clear */
+		this.numberInputField.setText("");
+		this.numberInputField.addKeyListener(new CustomKeyListener(
+				cell, gameSession));
+
+		if (!isRefresh) {
+			this.numberInputField.getHighlighter().removeAllHighlights();
+			this.numberInputField.addFocusListener(focusAdapter);
+		}
+
+		if (cell.hasNumber()) {
+			this.numberInputCell.setVisible(true);
+			this.pencilMarkDisplayCell.setVisible(false);
+			this.pencilMarkInputCell.setVisible(false);
+			this.numberInputField.setText(Integer.toString(cell.getNumber()));
+
+			if (!isRefresh) {
+				markGivenCell();
+				this.numberInputField.setEditable(false);
+				this.numberInputField.setFocusable(false);
+			}
+
+		} else {
+
+			this.pencilMarkDisplayCell.setVisible(false);
+			this.pencilMarkInputCell.setVisible(false);
+
+			if (cell.getPencilMarks().isEmpty()) {
+				this.numberInputCell.setVisible(true);
+				this.numberInputField.setEditable(true);
+				this.numberInputField.setFocusable(true);
+
+			} else {
+				cellLostFocus(null);
+			}
+		}
+	}
+
+	private void initPencilMarkInputCell() {
+		pencilMarkInputCell = new JPanel();

```

Comment:
same thing. since there is no pencilMarkInputCell local member, it doesn't
matter. but yet I will update them.

On Sun, Jul 12, 2015 at 10:24 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/CellGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34429881>:
>
> > +			this.pencilMarkDisplayCell.setVisible(false);
> > +			this.pencilMarkInputCell.setVisible(false);
> > +
> > +			if (cell.getPencilMarks().isEmpty()) {
> > +				this.numberInputCell.setVisible(true);
> > +				this.numberInputField.setEditable(true);
> > +				this.numberInputField.setFocusable(true);
> > +
> > +			} else {
> > +				cellLostFocus(null);
> > +			}
> > +		}
> > +	}
> > +
> > +	private void initPencilMarkInputCell() {
> > +		pencilMarkInputCell = new JPanel();
>
> this.pencilMarkInputCell
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34429881>.
>